src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1371,28 +1371,29 @@ protected virtual void Dispose(bool disposing)
             int dataCount,
             IntPtr data)
         {
+#if FEATURE_MANAGED_ETW || FEATURE_PERFTRACING
+            bool allAreNull = true;
 #if FEATURE_MANAGED_ETW
-            if (m_etwProvider == null)
+            allAreNull &= (m_etwProvider == null);
+            if (m_etwProvider != null
+                && !m_etwProvider.WriteEventRaw(ref eventDescriptor, eventHandle, activityID, relatedActivityID, dataCount, data))
             {
                 ThrowEventSourceException(eventName);
             }
-            else
-            {
-                if (!m_etwProvider.WriteEventRaw(ref eventDescriptor, eventHandle, activityID, relatedActivityID, dataCount, data))
-                    ThrowEventSourceException(eventName);
-            }
 #endif // FEATURE_MANAGED_ETW
 #if FEATURE_PERFTRACING
-            if (m_eventPipeProvider == null)
+            allAreNull &= (m_eventPipeProvider == null);
+            if (m_eventPipeProvider != null
+                && !m_eventPipeProvider.WriteEventRaw(ref eventDescriptor, eventHandle, activityID, relatedActivityID, dataCount, data))
             {
                 ThrowEventSourceException(eventName);
             }
-            else
+#endif // FEATURE_PERFTRACING
+            if (allAreNull)
             {
-                if (!m_eventPipeProvider.WriteEventRaw(ref eventDescriptor, eventHandle, activityID, relatedActivityID, dataCount, data))
-                    ThrowEventSourceException(eventName);
+                ThrowEventSourceException(eventName);
```

Comment:
Is there a reason that having all providers null would trigger an exception in this path but it is silently ignored in WriteEventString?

Reply:
I don't know the original intent, I was just keeping the logic the same as it was previously. If I had to guess the intention is something should have noticed above WriteEventRaw if the event is not enabled

Reply:
Knowing that WriteEventString() handles error cases it makes a little more sense, but I am still suspicious that the pre-existing logic was fishy. WriteEventString() also calls MultiMergeInner in some cases which calls WriteEventRaw so we wind up with different exception behavior for the errors on manifested EventSources vs. self-describing EventSources. I'm fine leaving it be if nobody is complaining though : )